Controllers/Frontend/PaymentPaypal.php

```diff
@@ -316,27 +308,23 @@ public function returnAction()
                         $redirectUrl .= '&token=' . urlencode($response['TOKEN']);
                         $this->redirect($redirectUrl);
                     } else {
-                        $this->redirect(
-                            array(
-                                'controller' => 'checkout',
-                                'action' => 'finish',
-                                'sUniqueID' => $response['PAYMENTREQUEST_0_CUSTOM']
-                            )
-                        );
+                        $this->forward('finish', 'checkout', 'frontend', array(
```

Comment:
We actually not want a "forward" here, otherwise the URL in the browser is not right.

Is there really no way reproduce this problem?
Perhaps a plugin is your fault.